src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py

```diff
@@ -330,6 +332,28 @@ def _configure_db_create_params(
     # needed, but we still need to avoid this conflict.
     c.argument('name', options_list=('--not-name',), arg_type=ignore_type)
 
+with ParametersContext(command='sql db show-connection-string') as c:
+    c.argument('client_provider',
+               options_list=('--client', '-c'),
+               help='Type of client connection provider.',
+               **enum_choice_list(ClientType))
+
+    auth_group = 'Authentication'
+
+    c.argument('auth_type',
+               options_list=('--auth-type', '-a'),
+               arg_group=auth_group,
+               help='Type of authentication.',
+               **enum_choice_list(ClientAuthenticationType))
+    c.argument('user',
+               options_list=('--user', '-u'),
+               arg_group=auth_group,
+               help='User name to substitute into the connection string.')
+    c.argument('password',
+               options_list=('--password', '-p'),
+               arg_group=auth_group,
+               help='Password to substitute into the connection string.')
```

Comment:
It is not a good practice for the user to input password on the command line because the bash history will record the password, hence, introduce attach vector. Therefore, it is generally discouraged for a command to ask the user to input password as a parameter (storage commands have the same issue, we're looking into approaches to address it as well.)

How about we remove this option, put a placeholder in the connection string and ask the user to replace it by themselves? It's up to you.

Reply:
That's better, also means we don't have to worry about escaping

Reply:
Plus, I believe the portal doesn't offer a connection string with password.

Reply:
Is not true. This is very useful feature when you are in CI/CD pipelines.
This would be helpful for me in Azure DevOps CLI, something like `az sql db show-connection-string -c ado.net ... -u $(MasterUser) -p $(MasterPassword)` (no history here).